src/core/generateZodSchema.ts

```diff
@@ -1407,20 +1524,31 @@ function buildSchemaReference(
 
   if (indexTypeName === "-1") {
     // Get the original type declaration
+    // For nested access like Superman["powers"][-1][-1], we need to find the root type
```

Comment:
Side note, we should support `Superman["powers"][number][number]` (same but different syntax)